code/datums/preferences.dm

```diff
@@ -500,16 +500,16 @@ datum/preferences
 				return TRUE
 
 			if ("update-age")
-				var/new_age = input(usr, "Please select type in age: 20-80", "Character Generation", src.age)  as null|num
+				var/new_age = tgui_input_number(usr, "Please select type in age: 20-80", "Character Generation", src.age, 80, 20)
 
 				if (new_age)
-					src.age = clamp(round(text2num(new_age)), 20, 80)
```

Comment:
I don't think TGUI guarantees the client won't return a wacky value, so probably best to leave the server side verification here.

Reply:
TGUI specifically /can't/ guarantee this